src/ui/public/utils/brush_event.js

```diff
@@ -1,13 +1,37 @@
 import moment from 'moment';
+import buildRangeFilter from 'ui/filter_manager/lib/range';
 export default function brushEventProvider(timefilter) {
-  return function (event) {
-    let from = moment(event.range[0]);
-    let to = moment(event.range[1]);
+  return function ($state) {
+    return function (event) {
+      if (event.data.xAxisField && event.data.xAxisField.type === 'date') {
+        let from = moment(event.range[0]);
+        let to = moment(event.range[1]);
 
-    if (to - from === 0) return;
+        if (to - from === 0) return;
 
-    timefilter.time.from = from;
-    timefilter.time.to = to;
-    timefilter.time.mode = 'absolute';
+        timefilter.time.from = from;
+        timefilter.time.to = to;
+        timefilter.time.mode = 'absolute';
+      } else if (event.data.xAxisField && event.data.xAxisField.type === 'number') {
+        let existingFilter = null;
+        $state.filters.forEach(function (it) {
+          if (it.meta && it.meta.key === event.data.xAxisField.name) {
+            existingFilter = it;
+          }
+        });
+
+        const range = {gte: event.range[0], lt: event.range[event.range.length - 1]};
+        if (existingFilter) {
+          existingFilter.range[event.data.xAxisField.name] = range;
+        } else {
+          const newFilter = buildRangeFilter(
+            event.data.xAxisField,
+            range,
+            event.data.indexPattern,
+            event.data.xAxisFormatter);
+          $state.$newFilters = [newFilter];
+        }
+      }
+    };
   };
-};
+};
```

Comment:
I think this might be a little easier to read if we made these changes:

* Add an early exit condition.
* Use a `switch`.
* Use the array `find` method instead of `forEach` (unless the forEach was intended to find the last matching filter?)

Here's what the changed code would look like:

```javascript
import moment from 'moment';
import buildRangeFilter from 'ui/filter_manager/lib/range';
export default function brushEventProvider(timefilter) {
  return $state => {
    return event => {
      if (!event.data.xAxisField) {
        return;
      }

      switch (event.data.xAxisField.type) {
        case 'date':
          let from = moment(event.range[0]);
          let to = moment(event.range[1]);

          if (to - from === 0) return;

          timefilter.time.from = from;
          timefilter.time.to = to;
          timefilter.time.mode = 'absolute';
          break;

        case 'number':
          const existingFilter = $state.filters.find(filter => (
            filter.meta && filter.meta.key === event.data.xAxisField.name
          ));

          const range = {gte: event.range[0], lt: event.range[event.range.length - 1]};
          if (existingFilter) {
            existingFilter.range[event.data.xAxisField.name] = range;
          } else {
            const newFilter = buildRangeFilter(
              event.data.xAxisField,
              range,
              event.data.indexPattern,
              event.data.xAxisFormatter);
            $state.$newFilters = [newFilter];
          }
          break;
      }
    };
  };
};
```